tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -973,6 +977,21 @@ public void expectCompletion(long timeoutMillis, String errorMsg) throws Interru
       } // else, ok
     }
 
+    public boolean tryExpectCompletion(long timeoutMillis) throws InterruptedException {
+      while (timeoutMillis-- > 0) {
+        Optional<T> value = abq.peek();
+        if (value != null) {
+          if (!value.isDefined()) {
+            abq.poll();
+            return true;
+          }
+          return false;
+        }
+        Thread.sleep(1);
```

Comment:
@akarnokd if you use abq.poll(remaining) instead of abq.peek() + abp.poll() you do not need to do the sleep(1)-dance.